test/unit/cloudwatch_agent/test_write_cloudwatch_agent_json.py

```diff
@@ -0,0 +1,209 @@
+    select_configs_for_feature,
+    select_configs_for_node_role,
+    select_configs_for_platform,
+    select_configs_for_scheduler,
+    select_metrics,
+)
+
+CONFIGS = [
+    {
+        "timestamp_format_key": "month_first",
+        "file_path": "/var/log/messages",
+        "log_stream_name": "system-messages",
+        "schedulers": ["awsbatch", "slurm"],
+        "node_roles": ["ComputeFleet", "HeadNode"],
+        "platforms": ["amazon", "centos"],
+        "feature_conditions": [{"dna_key": "dcv_enabled", "satisfying_values": ["head_node"]}],
+    },
+    {
+        "timestamp_format_key": "month_first",
+        "file_path": "/var/log/syslog",
+        "log_stream_name": "syslog",
+        "log_group_name": "pre-existing",
+        "schedulers": ["awsbatch", "slurm"],
+        "node_roles": ["ComputeFleet", "HeadNode"],
+        "platforms": ["ubuntu"],
+        "feature_conditions": [{"dna_key": ["directory_service", "enabled"], "satisfying_values": ["true"]}],
+    },
+    {
+        "timestamp_format_key": "default",
+        "file_path": "/var/log/cfn-init.log",
+        "log_stream_name": "cfn-init",
+        "schedulers": ["awsbatch"],
+        "node_roles": ["HeadNode"],
+        "platforms": ["amazon", "centos", "ubuntu"],
+        "feature_conditions": [],
+    },
+]
+
+METRIC_CONFIGS = {
+    "metrics_collected": [
+        {
+            "metric_type": "mem",
+            "measurement": ["mem_used_percent"],
+            "metrics_collection_interval": 60,
+            "node_roles": ["HeadNode"],
+        },
+        {
+            "metric_type": "disk",
+            "measurement": ["disk_used_percent"],
+            "resources": ["/"],
+            "metrics_collection_interval": 60,
+            "node_roles": ["HeadNode"],
+        },
+    ],
+    "append_dimensions": ["InstanceId"],
+    "aggregation_dimensions": [["InstanceId", "path"]],
+}
+
+
+@pytest.mark.asyncio
+def test_add_log_group_name_params():
+    configs = add_log_group_name_params("test", CONFIGS)
+    for config in configs:
+        assert_that(config).contains("log_group_name")
+        assert_that(config["log_group_name"]).is_equal_to("test")
+
+
+@pytest.mark.asyncio
+def test_add_instance_log_stream_prefixes(mocker):
+    instance_id = "i-0096test"
+    mocker.patch(
+        "write_cloudwatch_agent_json.gethostname",
+        return_value=instance_id,
+    )
+
+    configs = add_instance_log_stream_prefixes(CONFIGS)
+    for config in configs:
+        assert_that(config["log_stream_name"]).contains(instance_id)
+
+
+@pytest.mark.asyncio
+def test_select_configs_for_scheduler():
+    configs = select_configs_for_scheduler(CONFIGS, "slurm")
+    assert_that(len(configs)).is_equal_to(2)
+    configs = select_configs_for_scheduler(CONFIGS, "awsbatch")
+    assert_that(len(configs)).is_equal_to(3)
+
+
+@pytest.mark.asyncio
+def test_select_configs_for_node_role():
+    configs = select_configs_for_node_role(CONFIGS, "ComputeFleet")
```

Comment:
A way to avoid code duplication for this kind of tests is to have a parametrize, first param is platform and second param is expected_length

You can also think to unify `test_select_configs_for_scheduler`, `test_select_configs_for_platform` etc, having a single parameter called dimension, given that you're doing the same kind of assertion for all of them.